base/set.jl

```diff
@@ -3,13 +3,21 @@
 struct Set{T} <: AbstractSet{T}
     dict::Dict{T,Nothing}
 
-    Set{T}() where {T} = new(Dict{T,Nothing}())
-    Set{T}(s::Set{T}) where {T} = new(Dict{T,Nothing}(s.dict))
+    global unsafe_set(dict::Dict{T,Nothing}) where {T} = new{T}(dict)
 end
 
+Set{T}() where {T} = unsafe_set(Dict{T,Nothing}())
+Set{T}(s::Set{T}) where {T} = unsafe_set(Dict{T,Nothing}(s.dict))
 Set{T}(itr) where {T} = union!(Set{T}(), itr)
 Set() = Set{Any}()
 
+function Set{T}(s::KeySet) where {T}
```

Comment:
```suggestion
function Set{T}(s::KeySet{T, <:Dict{T}}) where {T}
```

Reply:
```suggestion
function Set{T}(s::KeySet{T}) where {T}
```
should be enough, shouldn’t it?